src/gc/gc.cpp

```diff
@@ -30484,6 +30484,9 @@ CObjectHeader* gc_heap::allocate_large_object (size_t jsize, int64_t& alloc_byte
 #endif //BACKGROUND_GC
 #endif // MARK_ARRAY
 
+    // these next few lines are not strictly necessary anymore - they are here
+    // to sanity check that we didn't get asked to create an object
+    // that's too large.
     size_t maxObjectSize = (INT32_MAX - 7 - Align(min_obj_size));
```

Comment:
I think you should be able to simplify it to:

```
#if BIT64
size_t maxObjectSize = (INT64_MAX - 7 - Align(min_obj_size)); 
#else
size_t maxObjectSize = (INT32_MAX - 7 - Align(min_obj_size)); 
#endif
```

And remove `GetGCAllowVeryLargeObjects`.